tests/run/default_args_introspection.pyx

```diff
@@ -0,0 +1,19 @@
+# mode: run
+# tag: default arguments
+
+cdef class C:
+    def f1(self, a, b=1, c=[]):
+        pass
+    def f2(self, a, b=1,/, c=[1]):
+        pass
+
+
+def check():
+    """
+    >>> import inspect
+    >>> inspect.signature(C.f1)
+    <Signature (self, a, b=1, c=[])>
+    >>> inspect.signature(C.f2)
+    <Signature (self, a, b=1, /, c=[1])>
```

Comment:
Wait and see, but I think this might fail on some of the earlier versions of Python that we still support (because inspect won't understand `/,`). It might be easier to check `C.__defaults__` and `C.__kwdefaults__` since this is what inspect uses.

If you want to include some tests with `inspect` then you might have to put some version checks in.

Reply:
This seems reasonable, I'll update test to use `__defaults__ ` and   `__kwdefaults__ `